rpc-server/src/dispatchers/blockchain.rs

```diff
@@ -454,7 +454,12 @@ impl BlockchainInterface for BlockchainDispatcher {
         let result = self.blockchain.read().get_account(&address);
 
         match result {
-            Some(account) => Ok(Account::from_account(address, account)),
+            Some(account) => match account {
+                nimiq_account::Account::Staking(_) => {
+                    Err(Error::GetAccountUnsupportedStakingContract)
+                }
+                _ => Ok(Account::from_account(address, account)),
+            },
```

Comment:
The better place to do this check in my opinion would be where the actual `unreachable!()` is. 
This change does not make it actually unreachable, once another place uses `Account::from_account` the caller needs to be aware to check this. 
I would much rather change `from_account` to return either `Option<Self>` or `Result<Self>`. Also note, that there are even more variants in the `Account` enum which would still reach `unreachable!()` with this change. Those are likely impossible to construct in this place, but that could change.